hoomd/mpcd/test/sphere_geometry_rejection_filler_mpi_test.cc

```diff
@@ -0,0 +1,100 @@
+        mpcd_snap->resize(1);
+
+        mpcd_snap->position[0] = vec3<Scalar>(1,-2,3);
+        mpcd_snap->velocity[0] = vec3<Scalar>(123, 456, 789);
+        }
+    auto mpcd_sys = std::make_shared<mpcd::SystemData>(mpcd_sys_snap);
+    auto pdata = mpcd_sys->getParticleData();
+    // There should be no virtual particles in the system at this point
+    UP_ASSERT_EQUAL(pdata->getNVirtual(), 0);
+    UP_ASSERT_EQUAL(pdata->getNVirtualGlobal(), 0);
+
+    // create a spherical confinement of radius 5.0
+    Scalar r=5.0;
+    auto sphere = std::make_shared<const mpcd::detail::SphereGeometry>(r, mpcd::detail::boundary::no_slip);
+    std::shared_ptr<::Variant> kT = std::make_shared<::VariantConst>(1.5);
+    std::shared_ptr<mpcd::RejectionVirtualParticleFiller<mpcd::detail::SphereGeometry>> filler = std::make_shared<F>(mpcd_sys, 2.0, 1, kT, 63, sphere);
+
+    /*
+     * Test basic filling up for this cell list
+     */
+    filler->fill(0);
+        {
+        ArrayHandle<Scalar4> h_pos(pdata->getPositions(), access_location::host, access_mode::read);
+        ArrayHandle<Scalar4> h_vel(pdata->getVelocities(), access_location::host, access_mode::read);
+        ArrayHandle<unsigned int> h_tag(pdata->getTags(), access_location::host, access_mode::read);
+        const BoxDim& box = sysdef->getParticleData()->getBox();
+        // check if the virtual particles are outside the sphere
+        unsigned int N_out(0);
+        for (unsigned int i = 0; i < pdata->getNVirtual(); ++i)
+            {
+            const unsigned int idx = pdata->getN() + i;
+            // check if the virtual particles are in the box
+            UP_ASSERT(h_pos.data[idx].x >= box.getLo().x && h_pos.data[idx].x < box.getHi().x);
+            UP_ASSERT(h_pos.data[idx].y >= box.getLo().y && h_pos.data[idx].y < box.getHi().y);
+            UP_ASSERT(h_pos.data[idx].z >= box.getLo().z && h_pos.data[idx].z < box.getHi().z);
+
+            Scalar3 pos = make_scalar3(h_pos.data[idx].x, h_pos.data[idx].y, h_pos.data[idx].z);
+            const Scalar r2 = dot(pos, pos);
+            if (r2 > r*r)
+                ++N_out;
+            }
+        UP_ASSERT_EQUAL(N_out, pdata->getNVirtual());
+        }
+
+    /*
+    * Test avg. number of virtual particles on each rank by filling up the system N_samples(=500) times
+    */
+    Scalar N_avg_rank(0);
+    Scalar N_avg_global(0);
+    unsigned int itr(500);
+    for (unsigned int t=0; t<itr; ++t)
+        {
+        pdata->removeVirtualParticles();
+        UP_ASSERT_EQUAL(pdata->getNVirtualGlobal(), 0);
+        filler->fill(1+t);
+
+        ArrayHandle<Scalar4> h_pos(pdata->getPositions(), access_location::host, access_mode::read);
+
+        N_avg_rank += pdata->getNVirtual();
+        N_avg_global += pdata->getNVirtualGlobal();
+        }
+    N_avg_rank /= itr;
+    N_avg_global /= itr;
+
+    // std::cout << "N_avg_rank = " << N_avg_rank << "\n";
+    UP_ASSERT_CLOSE(N_avg_rank, Scalar(1869), tol_small);
```

Comment:
It would be helpful to have a comment about where these target values are coming from